src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Changed.cs

```diff
@@ -118,4 +118,128 @@ public static void FileSystemWatcher_Changed_FileInNestedDirectory()
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        using (var dir1 = Utility.CreateTestDirectory(Path.Combine(dir.Path, "dir1")))
+        using (var dir2 = Utility.CreateTestDirectory(Path.Combine(dir1.Path, "dir2")))
+        using (var dir3 = Utility.CreateTestDirectory(Path.Combine(dir2.Path, "dir3")))
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+            byte[] bt = new byte[4096];
+
+            using (FileStream file = File.Create(Path.Combine(dir3.Path, "testfile.txt")))
+            {
+                // Attach the FSW to the existing structure
+                watcher.Path = Path.GetFullPath(dir.Path);
+                watcher.Filter = "*";
+                watcher.NotifyFilter = NotifyFilters.Attributes | NotifyFilters.CreationTime | NotifyFilters.DirectoryName | NotifyFilters.FileName | NotifyFilters.LastAccess | NotifyFilters.LastWrite | NotifyFilters.Security | NotifyFilters.Size;
+                watcher.IncludeSubdirectories = true;
+                watcher.EnableRaisingEvents = true;
+
+                // Change the nested file and verify we get the changed event
+                file.Write(bt, 0, bt.Length);
+                file.Flush();
+            }
+
+            Utility.ExpectEvent(eventOccured, "file changed", 10000);
+
+            // Stop the FSW
+            watcher.EnableRaisingEvents = false;
+
+            using (FileStream file2 = File.Create(Path.Combine(dir3.Path, "testfile2.txt")))
+            {
+                // Start the FSW, write to the file, expect to pick up the write
+                watcher.EnableRaisingEvents = true;
+                file2.Write(bt, 0, bt.Length);
+                file2.Flush();
+            }
+
+            Utility.ExpectEvent(eventOccured, "second file change");
+        }
+    }
+
+    [Fact, ActiveIssue("1477", PlatformID.Windows)]
+    public static void FileSystemWatcher_Changed_SymLinkFileDoesntFireEvent()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var file = Utility.CreateTestFile(Path.GetTempFileName()))
+        using (var watcher = new FileSystemWatcher())
+        {
+            // Create the symlink first
+            Utility.CreateSymLink(Path.GetFullPath(file.Path), Path.Combine(dir.Path, "link"), false);
+
+            AutoResetEvent are = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+
+            // Setup the watcher
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.NotifyFilter = NotifyFilters.FileName | NotifyFilters.Size;
+            watcher.EnableRaisingEvents = true;
+
+            // Changing the temp file should not fire an event through the symlink
+            byte[] bt = new byte[4096];
+            file.Write(bt, 0, bt.Length);
+            file.Flush();
```

Comment:
The previous test I believe required closing the file being written to in order to trigger the event; since we're not closing it here, seems like there could be a false negative: even if the component were incorrectly raising events for the linked file's changes, those changes wouldn't have actually occurred yet, so we won't pick up any on the symlink either.  Seems like you might actually want two different watchers for this test: one monitoring the target file and one monitoring the symlink, verifying that an event fires for the target but not for the symlink.